controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -443,6 +448,158 @@ bool master_thread::handle_cmdu_1905_autoconfiguration_WSC(Socket *sd,
+    std::list<sMacAddr> ruid_list;
+    // parse all tlvs in cmdu
+    int tlvType;
+    while ((tlvType = cmdu_rx.getNextTlvType()) != int(ieee1905_1::eTlvType::TLV_END_OF_MESSAGE)) {
+
+        if (tlvType < 0) {
+            LOG(ERROR) << "getNextTlvType has failed";
+            return false;
+        }
+
+        if (tlvType == int(wfa_map::eTlvTypeMap::TLV_CHANNEL_PREFERENCE)) {
+
+            // parse channel preference report message
+            auto channel_preference_tlv_rx = cmdu_rx.addClass<wfa_map::tlvChannelPreference>();
+            if (!channel_preference_tlv_rx) {
+                LOG(ERROR) << "addClass wfa_map::tlvChannelPreference has failed";
+                return false;
+            }
+
+            // read all operating class list
+            auto operating_classes_list_length =
+                channel_preference_tlv_rx->operating_classes_list_length();
+
+            for (int oc_idx = 0; oc_idx < operating_classes_list_length; oc_idx++) {
+                std::stringstream ss;
+                auto operating_class_tuple =
+                    channel_preference_tlv_rx->operating_classes_list(oc_idx);
+                if (!std::get<0>(operating_class_tuple)) {
+                    LOG(ERROR) << "getting operating class entry has failed!";
+                    return false;
+                }
+                auto &op_class_channels_rx = std::get<1>(operating_class_tuple);
+                auto operating_class       = op_class_channels_rx.operating_class();
+                ss << "operating class=" << int(operating_class);
+
+                const auto &flags = op_class_channels_rx.flags();
+                auto preference   = flags.preference;
+                auto reason_code  = flags.reason_code;
+                ss << ", preference=" << int(preference) << ", reason=" << int(reason_code);
+                ss << ", channel_list={";
+
+                auto channel_list_length = op_class_channels_rx.channel_list_length();
+                for (int ch_idx = 0; ch_idx < channel_list_length; ch_idx++) {
+                    auto channel_tuple = op_class_channels_rx.channel_list(ch_idx);
+                    if (!std::get<0>(channel_tuple)) {
+                        LOG(ERROR) << "getting channel entry has failed!";
+                        return false;
+                    }
+
+                    auto channel_rx = std::get<1>(channel_tuple);
+                    ss << int(channel_rx);
+
+                    // If reply tlvChannelPreference was not created for this ruid, then:
+                    // mark first supported non-dfs channel as selected channel
+                    // TODO: need to check that selected channel does not violate radio restriction
+                    const auto &ruid = channel_preference_tlv_rx->radio_uid();
+
+                    if (std::find(ruid_list.begin(), ruid_list.end(), ruid) == ruid_list.end() &&
```

Comment:
I prefer to reduce indentation by checking the opposite and continue instead of keeping the logic inside the if statement:
```
if (not what i'm looking for)
     continue;
// all the rest of the logic not indented
```

In general, try to reduce the `if if if if else if else else`s

Reply:
fixed